### ISSUE TITLE
Find better solution for timing issue when javascript functions are invoked via CEFSharp

### ISSUE THREAD

**Issue by alfarok** [alfarok/DynamoViewport#21]
For now 3 try/catch statements prevent this error from firing.  It seems most apparent on the last case `update` when a toggling of geometry happens.  It could be due to an overlap of event firing.  I think when a IsVisible event is triggered additional workspace events fire but the behavior is not consistent on show/hide.  See this comment for more details and convoluted logic:

https://github.com/alfarok/DynamoViewport/blob/master/src/ViewportViewExtension/ViewportViewExtension/ViewModels/ViewportWindowViewModel.cs#L150 

Example try/catch:

https://github.com/alfarok/DynamoViewport/blob/master/src/ViewportViewExtension/ViewportViewExtension/Resources/assests/js/viewer.js#L224

Old syncing issue:

#5

**Comment by alfarok** [comment]
Still seeing weird behavior when toggling visibility.  Could be related to graphs that have executed and have hidden nodes prior to Viewport launching.  Seems like force re-execution of the graph would help ensure the initial state as opposed to trying to handle all the states case by case initially.

Sample test graph:
![image](https://user-images.githubusercontent.com/13341935/91671572-f5350080-eaf5-11ea-9d6e-4b4c5b31e81d.png)
```json
{
  "Uuid": "4cf2fcd9-7c35-45e4-b1b2-533f51169a22",
  "IsCustomNode": false,
  "Description": null,
  "Name": "delete-2",
  "ElementResolver": {
    "ResolutionMap": {
      "Point": {
        "Key": "Autodesk.DesignScript.Geometry.Point",
        "Value": "ProtoGeometry.dll"
      },
      "Vector": {
        "Key": "Autodesk.DesignScript.Geometry.Vector",
        "Value": "ProtoGeometry.dll"
      }
    }
  },
  "Inputs": [],
  "Outputs": [],
  "Nodes": [
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "Point.ByCoordinates(5,5,0);",
      "Id": "ae9593df9ac943d783303003977d0ed6",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "88740e9274034b47a76cf49d78202aae",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "Point.ByCoordinates(100,5,0);",
      "Id": "dc0e727a92b94c219691697dd991a0a0",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "314ddb84fae7417f90a36325067029c0",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "Point.ByCoordinates(5,100,0);",
      "Id": "5d80fda776cf41a5afe569ae9f69073d",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "0a7b3b6e7d684fa9969bf6c2141d85f5",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Line.ByStartPointEndPoint@Autodesk.DesignScript.Geometry.Point,Autodesk.DesignScript.Geometry.Point",
      "Id": "46bd30f43d7140dd8fa1991049d403af",
      "Inputs": [
        {
          "Id": "11927d216cb7490e95e89e3406073f01",
          "Name": "startPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "f0e420127b5d4d8e9ca5379cc3f07241",
          "Name": "endPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "e781f2db52824d8887b1b3a122b4d927",
          "Name": "Line",
          "Description": "Line",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Creates a straight Line between two input Points.\n\nLine.ByStartPointEndPoint (startPoint: Point, endPoint: Point): Line"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Line.ByStartPointEndPoint@Autodesk.DesignScript.Geometry.Point,Autodesk.DesignScript.Geometry.Point",
      "Id": "1ebc87d319f74643a7d36446dc9816a5",
      "Inputs": [
        {
          "Id": "dd5293e1fda34d55b3d6a748689c9e21",
          "Name": "startPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "613642a65b27413997732f225df5fb45",
          "Name": "endPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "44c36aa13b074f44b601ca8fa0c29d6c",
          "Name": "Line",
          "Description": "Line",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Creates a straight Line between two input Points.\n\nLine.ByStartPointEndPoint (startPoint: Point, endPoint: Point): Line"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Line.ByStartPointEndPoint@Autodesk.DesignScript.Geometry.Point,Autodesk.DesignScript.Geometry.Point",
      "Id": "606f7943501e4fd7a14a24028cea0d51",
      "Inputs": [
        {
          "Id": "818e7ac83ae34d8383e70c14a6b00f59",
          "Name": "startPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "cde7f56eb8624b28b4ce256c1396cc16",
          "Name": "endPoint",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "7414b13881424031b17dd4cb48281549",
          "Name": "Line",
          "Description": "Line",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Creates a straight Line between two input Points.\n\nLine.ByStartPointEndPoint (startPoint: Point, endPoint: Point): Line"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.PolyCurve.ByJoinedCurves@Autodesk.DesignScript.Geometry.Curve[],double",
      "Id": "2b151a1ada0e426da200f653664d1be9",
      "Inputs": [
        {
          "Id": "47aff288141142e0bdc5aecd9b810e5f",
          "Name": "curves",
          "Description": "Curves to join into polycurve\n\nCurve[]",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "9a9d8b56e5b54685b182c718af70419f",
          "Name": "joinTolerance",
          "Description": "Tolerance to determine size of gap allowed between curves to be joined\n\ndouble\nDefault value : 0.001",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "e73633f4b515473f9e407d592cc2b107",
          "Name": "PolyCurve",
          "Description": "PolyCurve",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Make PolyCurve by joining curves. Flips curve as needed for connectivity. Choose a preferred join tolerance between 1e-6 and 1e-3 units.\n\nPolyCurve.ByJoinedCurves (curves: Curve[], joinTolerance: double = 0.001): PolyCurve"
    },
    {
      "ConcreteType": "CoreNodeModels.CreateList, CoreNodeModels",
      "VariableInputPorts": true,
      "NodeType": "ExtensionNode",
      "Id": "02636bcfded5441187cf0f9830bf1e91",
      "Inputs": [
        {
          "Id": "6da427921810463499dda04fcdcda5aa",
          "Name": "item0",
          "Description": "Item Index #0",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "5569bc3cada446d1a44a332b3c7b55b5",
          "Name": "item1",
          "Description": "Item Index #1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "467413736e0e4ffbb8f50eff32650774",
          "Name": "item2",
          "Description": "Item Index #2",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "865d342b0ae84c908e69f7c1281dd391",
          "Name": "list",
          "Description": "A list",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Makes a new list out of the given inputs"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Surface.ByPatch@Autodesk.DesignScript.Geometry.Curve",
      "Id": "b19719e5aab040bd80c870eb4763a5a3",
      "Inputs": [
        {
          "Id": "484237b3c04e40a1b2c947bcd56dfcb8",
          "Name": "closedCurve",
          "Description": "Curve",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "9111b90ccd5243618627f13fc097a8b3",
          "Name": "Surface",
          "Description": "Surface",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Create a Surface by filling in the interior of a closed boundary defined by input Curves.\n\nSurface.ByPatch (closedCurve: Curve): Surface"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Curve.Extrude@double",
      "Id": "3c857f5a14114cd091cda3a24995acb5",
      "Inputs": [
        {
          "Id": "090df92fecdf467398a4999220389f2b",
          "Name": "curve",
          "Description": "Autodesk.DesignScript.Geometry.Curve",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "5013cc6d1f22481db593f877a2292733",
          "Name": "distance",
          "Description": "The distance to extrude the curve\n\ndouble\nDefault value : 1",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "867d68f5ef9545fb94cdcea763ae5663",
          "Name": "Surface",
          "Description": "The extruded Surface",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Extrudes a Curve in the normal Vector direction\n\nCurve.Extrude (distance: double = 1): Surface"
    },
    {
      "ConcreteType": "CoreNodeModels.Input.IntegerSlider, CoreNodeModels",
      "NodeType": "NumberInputNode",
      "NumberType": "Integer",
      "InputValue": 71,
      "MaximumValue": 100,
      "MinimumValue": 0,
      "StepValue": 1,
      "Id": "72a7d02f4d584cd18fa59b0ba72603c4",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "8371ab3fe3164020a3b67c6539366c73",
          "Name": "",
          "Description": "Int64",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "A slider that produces integer values."
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Geometry.Rotate@Autodesk.DesignScript.Geometry.Point,Autodesk.DesignScript.Geometry.Vector,double",
      "Id": "60851535410748beb6988d2f9070a23d",
      "Inputs": [
        {
          "Id": "3af4ece786f84f3c9e0e4a312bd65dec",
          "Name": "geometry",
          "Description": "Autodesk.DesignScript.Geometry.Geometry",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "01b437092e544e2ba840345e063abda6",
          "Name": "origin",
          "Description": "Point",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "98d1ec2127a246fba9a8c544f8cb0850",
          "Name": "axis",
          "Description": "Vector",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "129b414d092f4e9ea7f7b44d80abdf8f",
          "Name": "degrees",
          "Description": "double\nDefault value : 0",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "db3f0e2c86ad4cc5a9f4d38e4147ce32",
          "Name": "Geometry",
          "Description": "Geometry",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Rotates an object around an origin and an axis by a specified degree\n\nGeometry.Rotate (origin: Point, axis: Vector, degrees: double = 0): Geometry"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Vector.ZAxis",
      "Id": "fddf8387c40c4c6794fc3af904725e02",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "21a310d4b536460db0c146cae9fd4467",
          "Name": "Vector",
          "Description": "Vector",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Get the canonical Z axis Vector (0,0,1)\n\nVector.ZAxis ( ): Vector"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "0..1080..90;",
      "Id": "896ee820f26b464a976b6d0df44aa7e0",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "291a5fd2763a4f6caed89a792cd67f0f",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Geometry.Translate@Autodesk.DesignScript.Geometry.Vector",
      "Id": "05b0eeea209748369cfe12c6999db118",
      "Inputs": [
        {
          "Id": "5e0568e9f9034b0d8cf525573ea60bda",
          "Name": "geometry",
          "Description": "Autodesk.DesignScript.Geometry.Geometry",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "becda032ad8d4545afdc9bae7714c600",
          "Name": "direction",
          "Description": "Vector",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "2f96fd8be9e24291af1c50383de5b02b",
          "Name": "Geometry",
          "Description": "Geometry",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Translate geometry in the given direction by the vector length\n\nGeometry.Translate (direction: Vector): Geometry"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "Vector.ByCoordinates(0,0,v);",
      "Id": "dc7ea45af243481886df9a01c29430df",
      "Inputs": [
        {
          "Id": "3c44b6afe1864acab7e0c8b672ec59f2",
          "Name": "v",
          "Description": "v",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "a892d761d87e41788975992be2eccb6b",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "0..100..#c;",
      "Id": "598902b601ec44db9b02aa46ff0c9634",
      "Inputs": [
        {
          "Id": "b8928b8fbeb240f1b54053e41ee2658d",
          "Name": "c",
          "Description": "c",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "4d61d17550ae4fcd9c4895913bddf5ab",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "DSCore.List.Count@var[]..[]",
      "Id": "34ab97b85d0744849210a3a301743a3c",
      "Inputs": [
        {
          "Id": "feed286cc66d40729c017fdd25901d56",
          "Name": "list",
          "Description": "List to get the item count of.\n\nvar[]..[]",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "ffdc71fbcee940d9a6c98457502afced",
          "Name": "count",
          "Description": "List length.",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Returns the number of items stored in the given list.\n\nList.Count (list: var[]..[]): int"
    }
  ],
  "Connectors": [
    {
      "Start": "88740e9274034b47a76cf49d78202aae",
      "End": "dd5293e1fda34d55b3d6a748689c9e21",
      "Id": "6cae61fd1fdc4d4bb066bd50ca0bacda"
    },
    {
      "Start": "88740e9274034b47a76cf49d78202aae",
      "End": "818e7ac83ae34d8383e70c14a6b00f59",
      "Id": "be6ee661978b4e1999f7b3ee5d2783d5"
    },
    {
      "Start": "88740e9274034b47a76cf49d78202aae",
      "End": "01b437092e544e2ba840345e063abda6",
      "Id": "4c424275396d4b9aa8beab2c2102bfa6"
    },
    {
      "Start": "314ddb84fae7417f90a36325067029c0",
      "End": "11927d216cb7490e95e89e3406073f01",
      "Id": "18d1960915504a77bd3c7ef0ca1b0531"
    },
    {
      "Start": "314ddb84fae7417f90a36325067029c0",
      "End": "613642a65b27413997732f225df5fb45",
      "Id": "cf24f20721bb419b8ac8889ccc1920a6"
    },
    {
      "Start": "0a7b3b6e7d684fa9969bf6c2141d85f5",
      "End": "f0e420127b5d4d8e9ca5379cc3f07241",
      "Id": "7fdb66afd63e4a37a95e3d51e5a8b106"
    },
    {
      "Start": "0a7b3b6e7d684fa9969bf6c2141d85f5",
      "End": "cde7f56eb8624b28b4ce256c1396cc16",
      "Id": "746c6a14f9fc495b9b6abe99186f1523"
    },
    {
      "Start": "e781f2db52824d8887b1b3a122b4d927",
      "End": "5569bc3cada446d1a44a332b3c7b55b5",
      "Id": "6c6f798abc2a4fc6b329f8e76c86c41e"
    },
    {
      "Start": "44c36aa13b074f44b601ca8fa0c29d6c",
      "End": "6da427921810463499dda04fcdcda5aa",
      "Id": "60096fbd5b57466fb82571524f065269"
    },
    {
      "Start": "7414b13881424031b17dd4cb48281549",
      "End": "467413736e0e4ffbb8f50eff32650774",
      "Id": "7cef125760054ef7bb659819270f954e"
    },
    {
      "Start": "e73633f4b515473f9e407d592cc2b107",
      "End": "484237b3c04e40a1b2c947bcd56dfcb8",
      "Id": "04ba9a05158f4a5c81962404a60a7ba3"
    },
    {
      "Start": "e73633f4b515473f9e407d592cc2b107",
      "End": "090df92fecdf467398a4999220389f2b",
      "Id": "aa2a9970afba4cbcb1237e7594e1276c"
    },
    {
      "Start": "865d342b0ae84c908e69f7c1281dd391",
      "End": "47aff288141142e0bdc5aecd9b810e5f",
      "Id": "b004fa656abf4937bd9a5418b959b754"
    },
    {
      "Start": "9111b90ccd5243618627f13fc097a8b3",
      "End": "3af4ece786f84f3c9e0e4a312bd65dec",
      "Id": "a4d373db6812474da9edba62512c8afb"
    },
    {
      "Start": "8371ab3fe3164020a3b67c6539366c73",
      "End": "5013cc6d1f22481db593f877a2292733",
      "Id": "4d821391db7848b4a12a59d5781c9849"
    },
    {
      "Start": "db3f0e2c86ad4cc5a9f4d38e4147ce32",
      "End": "5e0568e9f9034b0d8cf525573ea60bda",
      "Id": "3875a5ab4560433c8fae26d24a5886f5"
    },
    {
      "Start": "21a310d4b536460db0c146cae9fd4467",
      "End": "98d1ec2127a246fba9a8c544f8cb0850",
      "Id": "cf3748562af34ec8a73e23be0278c128"
    },
    {
      "Start": "291a5fd2763a4f6caed89a792cd67f0f",
      "End": "129b414d092f4e9ea7f7b44d80abdf8f",
      "Id": "d56a7f18339c494099e74d87f3995cb9"
    },
    {
      "Start": "291a5fd2763a4f6caed89a792cd67f0f",
      "End": "feed286cc66d40729c017fdd25901d56",
      "Id": "adf55392d7c346c4a7c6f16e3e5675b2"
    },
    {
      "Start": "a892d761d87e41788975992be2eccb6b",
      "End": "becda032ad8d4545afdc9bae7714c600",
      "Id": "d5fa6d95a26f4883910a6f22533a4e61"
    },
    {
      "Start": "4d61d17550ae4fcd9c4895913bddf5ab",
      "End": "3c44b6afe1864acab7e0c8b672ec59f2",
      "Id": "607d09f77ead48b096b7b0b8f4e10f24"
    },
    {
      "Start": "ffdc71fbcee940d9a6c98457502afced",
      "End": "b8928b8fbeb240f1b54053e41ee2658d",
      "Id": "1f76284857a94b8c9e3b977cb19fd69f"
    }
  ],
  "Dependencies": [],
  "NodeLibraryDependencies": [],
  "Bindings": [],
  "View": {
    "Dynamo": {
      "ScaleFactor": 1.0,
      "HasRunWithoutCrash": true,
      "IsVisibleInDynamoLibrary": true,
      "Version": "2.9.0.2416",
      "RunType": "Automatic",
      "RunPeriod": "1000"
    },
    "Camera": {
      "Name": "Background Preview",
      "EyeX": -17.0,
      "EyeY": 24.0,
      "EyeZ": 50.0,
      "LookX": 12.0,
      "LookY": -13.0,
      "LookZ": -58.0,
      "UpX": 0.0,
      "UpY": 1.0,
      "UpZ": 0.0
    },
    "NodeViews": [
      {
        "Id": "ae9593df9ac943d783303003977d0ed6",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 189.03857566765578,
        "Y": 103.15430267062311
      },
      {
        "Id": "dc0e727a92b94c219691697dd991a0a0",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 176.80712166172123,
        "Y": 197.93175074183964
      },
      {
        "Id": "5d80fda776cf41a5afe569ae9f69073d",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 173.95845697329395,
        "Y": 305.08308605341233
      },
      {
        "Id": "46bd30f43d7140dd8fa1991049d403af",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Line.ByStartPointEndPoint",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 583.85163204747744,
        "Y": 202.12462908011872
      },
      {
        "Id": "1ebc87d319f74643a7d36446dc9816a5",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Line.ByStartPointEndPoint",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 581.15430267062311,
        "Y": 86.635014836795364
      },
      {
        "Id": "606f7943501e4fd7a14a24028cea0d51",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Line.ByStartPointEndPoint",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 584.89614243323422,
        "Y": 320.61127596439178
      },
      {
        "Id": "2b151a1ada0e426da200f653664d1be9",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "PolyCurve.ByJoinedCurves",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 968.42136498516311,
        "Y": 178.38575667655778
      },
      {
        "Id": "02636bcfded5441187cf0f9830bf1e91",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "List Create",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 827.17507418397611,
        "Y": 178.38575667655783
      },
      {
        "Id": "b19719e5aab040bd80c870eb4763a5a3",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Surface.ByPatch",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1253.2878338278929,
        "Y": 178.38575667655772
      },
      {
        "Id": "3c857f5a14114cd091cda3a24995acb5",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Curve.Extrude",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1267.5311572700302,
        "Y": 297.1419361674603
      },
      {
        "Id": "72a7d02f4d584cd18fa59b0ba72603c4",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Integer Slider",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 911.44807121661745,
        "Y": 327.94065281899105
      },
      {
        "Id": "60851535410748beb6988d2f9070a23d",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Geometry.Rotate",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1602.0761454127178,
        "Y": 451.06629057127077
      },
      {
        "Id": "fddf8387c40c4c6794fc3af904725e02",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Vector.ZAxis",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1461.6285698252186,
        "Y": 503.29180985910455
      },
      {
        "Id": "896ee820f26b464a976b6d0df44aa7e0",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1402.7142946458785,
        "Y": 587.85305839887167
      },
      {
        "Id": "05b0eeea209748369cfe12c6999db118",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Geometry.Translate",
        "ShowGeometry": true,
        "Excluded": false,
        "X": 2278.0156437292353,
        "Y": 643.97870193421147
      },
      {
        "Id": "dc7ea45af243481886df9a01c29430df",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1939.7330636665738,
        "Y": 694.82351044186271
      },
      {
        "Id": "598902b601ec44db9b02aa46ff0c9634",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1759.7285750742021,
        "Y": 695.64251248861524
      },
      {
        "Id": "34ab97b85d0744849210a3a301743a3c",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "List.Count",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 1570.0618934882079,
        "Y": 693.73588557446067
      }
    ],
    "Annotations": [],
    "X": -63.450987177172465,
    "Y": 73.994718468062956,
    "Zoom": 0.60906559778322589
  }
}
```

**Comment by alfarok** [comment]
This was caused by modifying the scene while it was being traversed.  Fixed in 009d402af393ab4caf0f2a79fed18921ba2557c4.  Considering reverting to the old logic of retrieving existing node by name as opposed to traversing now that it seems like the viewer and graph are sync'd and no longer displaying symptoms of a race condition.  It would be significantly more performative pending there is not a race condition from the CEFSharp bridge.

**Comment by alfarok** [comment]
stop traversing the scene for a single obj w/ unique name: 7188f0ce2de11ab188d3e2c2c5d25659580d2e33
